### PR TITLE
8251241: macOS: iconify property doesn't change after minimize when resizable is false

### PR DESCRIPTION
**Pull request by FlorianKirmaier**
ticket: https://bugs.openjdk.java.net/browse/JDK-8251241

This small change fixing the minimize for mac.
Changes are property registered.
Calling setIconize(true) no longer resets to false.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8251241](https://bugs.openjdk.java.net/browse/JDK-8251241): macOS: iconify property doesn't change after minimize when resizable is false

### Reviewers
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)
 * Ambarish Rapte ([arapte](@arapte) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/280/head:pull/280`
`$ git checkout pull/280`

### CODE REVIEW

**Comment by kevinrushforth** [comment]
> JDK-8251241: macOS: iconify property doesn't change after minimize when resizable is false ⚠️ Title mismatch between PR and JBS.

Can you change the title to match the JBS title?

**Comment by FlorianKirmaier** [comment]
Done! I've changed the title.

**Comment by dlemmermann** [comment]
Thank you Florian!!! :-)

File: modules/javafx.graphics/src/main/native-glass/mac/GlassWindow+Java.m [hunk 1]
```diff
@@ -94,10 +94,14 @@ - (void)_sendJavaWindowResizeEvent:(int)type forFrame:(NSRect)frame
...
     {
         GET_MAIN_JENV;
-        (*env)->CallVoidMethod(env, jWindow, jWindowNotifyResize,
-            [self->nsWindow isZoomed]
-                ? com_sun_glass_events_WindowEvent_MAXIMIZE
-                : type,
+
+        if([self->nsWindow isMiniaturized]) {
+            type = com_sun_glass_events_WindowEvent_MINIMIZE;
+        } else if([self->nsWindow isZoomed]) {
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/native-glass/mac/GlassWindow+Java.m:100]: Minor: can you add a space between `if` and `(` to match our coding style?

Comment (FlorianKirmaier) [reply modules/javafx.graphics/src/main/native-glass/mac/GlassWindow+Java.m]: Done!

**Comment by FlorianKirmaier** [comment]
Thank you for the very quick feedback!
In the application, the fix was developed for, it also seems to fix all issues related to iconified.

**Comment by kevinrushforth** [comment]
You may have seen my comment in JBS: This PR also fixes [JDK-8249202](https://bugs.openjdk.java.net/browse/JDK-8249202), which I discovered while testing [JDK-8248490](https://bugs.openjdk.java.net/browse/JDK-8248490). This means that I already have an automated test that can be modified to test this. I just need to add a couple things to it (an assertion check for the iconified property, a check to ensure that we can deiconify the stage, and then add a mode to run it with a non-resizable stage). Once I verify it, I can send you the patch for the test, if that's OK with you?

**Comment by kevinrushforth** [comment]
Here is [the patch](https://github.com/openjdk/jfx/files/5071008/icontest.patch.txt) for [IconifyTest.java](https://github.com/openjdk/jfx/blob/1f42cebed04d88998d1a66ce99ba3c51995a18a8/tests/system/src/test/java/test/robot/javafx/stage/IconifyTest.java) :

[icontest.patch.txt](https://github.com/openjdk/jfx/files/5071008/icontest.patch.txt)

I verified that it catches the bugs, in that the modified test fails without the fix and passes with the fix.

Can you apply the patch to your repo and push a new commit with the updated test?

**Comment by FlorianKirmaier** [comment]
I've added your patch for the unit-test!
Nice to see it fixes more bugs. : )
Don't want to know how many hours were spent working around this bug by all the people around the world.
Compared to this, the fix was developed really quickly.

**Comment by kevinrushforth** [comment]
@arapte can you be a second reviewer?

**Review by arapte — APPROVED** [review]

**Pull request closed without merging**